src/fenicsxconcrete/experimental_setup/compression_cylinder.py

```diff
@@ -77,6 +78,13 @@ def generate_cylinder_mesh(radius:float, height:float, mesh_density:float, eleme
     with df.io.XDMFFile(MPI.COMM_WORLD,xdmf_file,"r") as mesh_file:
         mesh = mesh_file.read_mesh(name="Grid")
 
+    # delete temp files
+    if folder_name.is_dir():
+        os.remove(msh_file)
+        os.remove(xdmf_file)
+        os.remove(xdmf_file.with_suffix('.h5'))
+        os.rmdir(folder_name)
+
```

Comment:
just a commend. you could use [`tempfile`](https://docs.python.org/3/library/tempfile.html) for generating temporary files and directories

Reply:
Thank you for the idea. In general I agree this would be better.
I had a quick look and could not find a suitable example of how to implement is in this specific case, where files are being created e.g. by dolfinx.
If someone feels strongly about this, I would suggest to do this in a new issue.

Reply:
If it's okay with you, I would add this here (later today). if we can add it here directly, we do not have to open another issue and I would prefer that.

Reply:
Sure, if you are motivated to do this today, feel free :)
I am curious to learn how to do this properly.

Reply:
it would be the same as you did, but just more convenient with `tempfile`, although it might be possible to use `dolfinx.io.gmsh.read_from_msh` instead of the conversion using `meshio` and then we can get the mesh and meshtags directly from the `msh`.

Reply:
hi @eriktamsen,
the mesh is created and read in on one process.
If you think that it's important that the mesh is also read in in parallel then we need to use `meshio` here. 
Let me know and I can change it.

Reply:
nice.
the "detour" using the external mesh file, was just because at that time I did not find a better solution.
the easier the better.